models/deviceresource.go

```diff
@@ -30,29 +31,24 @@ type DeviceResource struct {
 // MarshalJSON implements the Marshaler interface in order to make empty strings null
 func (do DeviceResource) MarshalJSON() ([]byte, error) {
 	test := struct {
-		Description *string           `json:"description,omitempty"`
-		Name        *string           `json:"name,omitempty"`
-		Tag         *string           `json:"tag,omitempty"`
-		Properties  ProfileProperty   `json:"properties"`
-		Attributes  map[string]string `json:"attributes,omitempty"`
+		Description string             `json:"description,omitempty"`
+		Name        string             `json:"name,omitempty"`
+		Tag         string             `json:"tag,omitempty"`
+		Properties  *ProfileProperty   `json:"properties,omitempty"`
+		Attributes  *map[string]string `json:"attributes,omitempty"`
```

Comment:
Why was this converted to a pointer?  omitempty tag will not include a map if it's empty.